vcd/resource_vcd_network_pool.go

```diff
@@ -0,0 +1,507 @@
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"transport_zone": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							MaxItems:    1,
+							Description: "Transport Zone Backing",
+							Elem:        resourceNetworkPoolBacking("resource"),
+						},
+						"port_groups": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							MaxItems:    1,
+							Description: "Backing port groups",
+							Elem:        resourceNetworkPoolBacking("resource"),
+						},
+						"distributed_switches": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
+							MaxItems:    1,
+							Description: "Backing distributed switches",
+							Elem:        resourceNetworkPoolBacking("resource"),
+						},
+						"range_ids": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Distributed Switch ID ranges (used with VLAN)",
+							Elem:        resourceNetworkPoolVlanIdRange,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceNetworkPoolCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	networkPoolName := d.Get("name").(string)
+	networkPoolDescription := d.Get("description").(string)
+
+	networkPoolType := d.Get("type").(string)
+	networkPoolProviderId := d.Get("network_provider_id").(string)
+	networkPoolProviderType := networkProviderVcenter
+	if strings.Contains(networkPoolProviderId, "urn:vcloud:nsxtmanager") {
+		networkPoolProviderType = networkProviderNsxtManager
+	}
+
+	var networkPoolProvider types.OpenApiReference
+
+	if networkPoolProviderType == networkProviderVcenter {
+		vCenter, err := vcdClient.GetVCenterById(networkPoolProviderId)
+		if err != nil {
+			return diag.Errorf("[network pool create] error retrieving vCenter with ID '%s': %s", networkPoolProviderId, err)
+		}
+		networkPoolProvider.Name = vCenter.VSphereVCenter.Name
+		networkPoolProvider.ID = vCenter.VSphereVCenter.VcId
+	} else {
+		managers, err := vcdClient.QueryNsxtManagers()
+		if err != nil {
+			return diag.Errorf("[network pool create] error retrieving list of NSX-T managers: %s", err)
+		}
+		var manager *types.QueryResultNsxtManagerRecordType
+
+		bareId := extractUuid(networkPoolProviderId)
+		for _, m := range managers {
+			if bareId == extractUuid(m.HREF) {
+				manager = m
+				break
+			}
+		}
+		if manager == nil {
+			return diag.Errorf("[network pool create] NSX-T manager with ID '%s' not found", networkPoolProviderId)
+		}
+		networkPoolProvider.Name = manager.Name
+		networkPoolProvider.ID = "urn:vcloud:nsxtmanager:" + extractUuid(manager.HREF)
+	}
+
+	if networkPoolProvider.ID == "" {
+		return diag.Errorf("[network pool create] no suitable network provider (%s) found from ID '%s'", networkPoolProviderType, networkPoolProviderId)
```

Comment:
Just wondering whether it is possible to reach this line? It seems `networkPoolProvider.ID` will always have a value?